### ISSUE TITLE
Create separate client/server processes for brain and world 

### ISSUE THREAD

**Issue by brohrer** [brohrer/becca#27]
Becca is intended for controlling physical robots. Currently the simulated world and brain alternate within the same loop in the same python process. This is unlikely to be the case in a physical robot implementation. This task to break the brain and world into two processes, along with all the coordination and communication that entails.

**Recommendations:**
The Robot Operating System ([ROS](http://www.ros.org/)) was designed to do exactly this. Dustin Franklin [implemented it](https://github.com/brohrer/becca_ros) for Becca four years ago, but Becca (and probably ROS) have changed so much since then as to be unrecognizeable. Feel free to use Dustin's work as a starting point if it's helpful.

As with all Becca work, please emphasize simplicity, clarity, and robustness over performance and efficiency. Minimize dependencies. Document generously.

**Completion**
Let's consider this issue closed when there is a non-trivial demo of the beca learning a simulated world in a client/server relationship.

**Comment by brohrer** [comment]
@markroxor feel free to dive in if this fits your interests.

**Comment by markroxor** [comment]
I see that installation instructions for [ROS Hydro](http://wiki.ros.org/hydro/Installation/Ubuntu) are ages old. I cannot install it for Ubuntu 18.04. :(

**Comment by brohrer** [comment]
Yes, it's true. It will probably require starting from scratch to get it going with the latest version of ROS. The original instructions might be helpful, but they may not.

Sorry! It's not a lot of guidance, but it's the best I have for now.
